hip-config.cmake.in

```diff
@@ -138,19 +138,23 @@ else()
 endif()
 
 if(HIP_COMPILER STREQUAL "clang")
-  set_property(TARGET hip::device APPEND PROPERTY
-    INTERFACE_COMPILE_OPTIONS -x hip --hip-device-lib-path=${AMD_DEVICE_LIBS_PREFIX}/lib
-  )
-
   if (HIP_CXX_COMPILER MATCHES ".*clang\\+\\+")
     set_property(TARGET hip::device APPEND PROPERTY
       INTERFACE_COMPILE_OPTIONS -mllvm -amdgpu-early-inline-all=true -mllvm -amdgpu-function-calls=false
     )
   endif()
 
-  set_property(TARGET hip::device APPEND PROPERTY
-    INTERFACE_LINK_LIBRARIES --hip-device-lib-path=${AMD_DEVICE_LIBS_PREFIX}/lib --hip-link
```

Comment:
we do not need --hip-device-lib-path in link step but we still need --hip-link, because for -fgpu-rdc case, --hip-link is needed to link device code from different compilation units together.

Reply:
How will the compiler find the path to the device libraries? It doesnt have access to CMAKE_PREFIX_PATH.

Reply:
find_dependency(AMDDeviceLibs)
sets AMD_DEVICE_LIBS_PREFIX

Reply:
Yes, but how can the compiler read the `AMD_DEVICE_LIBS_PREFIX` variable? Its set by cmake.

Reply:
they are in device lib config cmake file, and loaded by find_dependency

Reply:
What do you mean "loaded by find_dependency"? I still don't understand how the compiler can know the path to the device libs unless we pass the path to the compiler(using the `AMD_DEVICE_LIBS_PREFIX` cmake variable). Clang will not call cmake and find_dependency to find the device libs.

Reply:
AMD_DEVICE_LIBS_PREFIX is not passed by user to cmake. It is defined in device lib cmake configure file. When device library is installed, its cmake config file is installed to /opt/rocm/lib/cmake/AMDDeviceLibs/AMDDeviceLibsConfig.cmake, which contains 

# Derive absolute install prefix from config file path.
get_filename_component(AMD_DEVICE_LIBS_PREFIX "${CMAKE_CURRENT_LIST_FILE}" PATH)
get_filename_component(AMD_DEVICE_LIBS_PREFIX "${AMD_DEVICE_LIBS_PREFIX}" PATH)
get_filename_component(AMD_DEVICE_LIBS_PREFIX "${AMD_DEVICE_LIBS_PREFIX}" PATH)
get_filename_component(AMD_DEVICE_LIBS_PREFIX "${AMD_DEVICE_LIBS_PREFIX}" PATH)

When an app does its own cmake and its CMakeList.txt contains find_package(HIP, config), cmake will load HIP cmake config file, which in turn loads device lib cmake config file, and  AMD_DEVICE_LIBS_PREFIX gets defined, then hip::device gets proper INTERFACE_COMPILE_OPTIONS which contains --hip-device-lib= option. Then when clang is called with correct compile option.

Reply:
>  then hip::device gets proper INTERFACE_COMPILE_OPTIONS which contains --hip-device-lib= option.

Ah yes, but doesn't this needed to be added while linking as well? Or does the compiler not read the the device libs during linking?

Reply:
linking of the device library is done at compilation stage for each compilation unit with special option. Device library is not needed in linking stage.